core/services/transmission/integration_test.go

```diff
@@ -0,0 +1,500 @@
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/solidity_vrf_consumer_interface_v08"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/vrf_coordinator_mock"
+
+	"github.com/ethereum/go-ethereum/core/types"
+
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/transmission/generated/entry_point"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/transmission/generated/greeter_wrapper"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/transmission/generated/paymaster_wrapper"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/transmission/generated/sca_wrapper"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/transmission/generated/smart_contract_account_factory"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/transmission/generated/smart_contract_account_helper"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/cltest"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/services/transmission"
+)
+
+var (
+	greeterABI    = evmtypes.MustGetABI(greeter_wrapper.GreeterABI)
+	consumerABI   = evmtypes.MustGetABI(solidity_vrf_consumer_interface_v08.VRFConsumerABI)
+	entrypointABI = evmtypes.MustGetABI(entry_point.EntryPointABI)
+)
+
+func Test4337Basic(t *testing.T) {
+	// Create a key for holder1 that we can use to sign
+	holder1Key := cltest.MustGenerateRandomKey(t)
+	t.Log("Holder key:", holder1Key.String())
+
+	// Construct simulated blockchain environmnet.
+	holder1Transactor, err := bind.NewKeyedTransactorWithChainID(holder1Key.ToEcdsaPrivKey(), testutils.SimulatedChainID)
+	require.NoError(t, err)
+	var (
+		metaERC20Owner = testutils.MustNewSimTransactor(t)
+		holder1        = holder1Transactor
+		holder2        = testutils.MustNewSimTransactor(t)
+		relay          = testutils.MustNewSimTransactor(t)
+	)
+	genesisData := core.GenesisAlloc{
+		metaERC20Owner.From: {Balance: assets.Ether(1000).ToInt()},
+		holder1.From:        {Balance: assets.Ether(1000).ToInt()},
+		holder2.From:        {Balance: assets.Ether(1000).ToInt()},
+		relay.From:          {Balance: assets.Ether(1000).ToInt()},
+	}
+	gasLimit := uint32(30e6)
+	backend := cltest.NewSimulatedBackend(t, genesisData, gasLimit)
+	backend.Commit()
+
+	// Deploy Entry Point, and Smart Contract Account factory and helper.
+	entryPointAddress, _, entryPoint, err := entry_point.DeployEntryPoint(holder1, backend)
+	require.NoError(t, err)
+	factoryAddress, _, _, _ := smart_contract_account_factory.DeploySmartContractAccountFactory(holder1, backend)
+	require.NoError(t, err)
+	_, _, helper, err := smart_contract_account_helper.DeploySmartContractAccountHelper(holder1, backend)
+	require.NoError(t, err)
+	greeterAddress, _, greeter, err := greeter_wrapper.DeployGreeter(holder1, backend)
+	require.NoError(t, err)
+	backend.Commit()
```

Comment:
Minor: this could be extracted into a `createTransmissionUniverse` helper and can be used in all subsequent tests.